models/user.js

```diff
@@ -0,0 +1,153 @@
+let uuid = require('uuid/v4');
+let models = require('./index');
+
+/**
+ * @class User
+ * @property {string} id
+ * @property {string} name
+ * @property {string} dob
+ * @property {string} address
+ * @property {string} description
+ * @property {string} createdAt
+ * @property {string} updatedAt
+ */
+class User {
+	/**
+	 * @param {User|string|object} [o]
+	 */
+	constructor(o) {
+		this.id = uuid();
+		this.name = undefined;
+		this.dob = undefined;
+		this.address = undefined;
+		this.description = undefined;
+		this.createdAt = undefined;
+		this.updatedAt = undefined;
+
+		if (!o) return;
+		if (typeof o === 'string') {
+			o = JSON.parse(o);
+		}
+		for (let k of Object.keys(this)) {
+			if (o[k] === undefined) {
+				this[k] = undefined;
+				continue;
+			}
+
+			this[k] = JSON.parse(JSON.stringify(o[k]));
```

Comment:
Yeah I hear that too.  Are you thinking Object.assign() for a shallow copy? I just err on the side of this strategy to avoid situations where something changes and now I've got shared references to deeply nested objects.

Were you thinking Object.assign() here?

Reply:
In this case simple assignment would work perfectly fine. Maybe if we really add nested objects I’d add special case for it and clone only what’s necesary. Object.assign does shallow copy so it’s probably not that usefull. Stringify/parse pattern, while horribly ugly, is still the most performant way of doing deep clone of simple objects :-)